### ISSUE TITLE
Hide apps and per app icon change (Request)

### ISSUE THREAD

**Issue by gcladores** [amirzaidi/Launcher3#472]
Hide apps and per app icon change from internal storage.

**Comment by amirzaidi** [comment]
Hiding apps is already in the launcher

**Comment by gcladores** [comment]
Yeah you're right. Sorry bout that. I was looking for it in the settings. What about force round icons?

**Comment by SANJAYSKIM** [comment]
How to hide??
In v3.5, app long press not showing options for this

**Comment by amirzaidi** [comment]
@gcladores force round icons is always enabled, legacy icon masking is disabled

**Comment by gcladores** [comment]
Google News and Weather, Google Chrome. VLC and other apps won't force round. google chrome has the white border. looks better when its just the chrome icon no white border. when I use Rootless Pixel 2 Launcher v4.0.3 from paphonb. the all have rounded icons. i switched because i don't like the pixel launcher android p dev preview

![screenshot_20180424-021106](https://user-images.githubusercontent.com/30974738/39145170-82ad81a8-4765-11e8-9d5d-b62ae69bd143.png)
![screenshot_20180424-021114](https://user-images.githubusercontent.com/30974738/39145183-890a841a-4765-11e8-8610-2f4cd7b6e5af.png)

**Comment by gcladores** [comment]
using Google Nexus 6P unrooted. Stock Android 8.1. from OTA. Substratum Themed.

**Comment by amirzaidi** [comment]
Substratum or any other OMS theme engine breaks the round icon extraction due to Android limitations

**Comment by wrick17** [comment]
I am not able to hide apps in the app drawer.

**Comment by perceival** [comment]
Hiding apps works. You need to long press icon in the drawer and choose a pencil. I was switching from another implementation of Pixel Launcher where there was an option in settings for it.
However... now how can I unhide the app?

**Comment by SANJAYSKIM** [comment]
Long press not showing pencil icon

**Comment by perceival** [comment]
I can say for Oreo only. Not sure which version you have.

**Comment by wrick17** [comment]
Hi, I'm on Oreo, long press does not get me the pencil icon. Only got the info icon.

**Comment by gcladores** [comment]
you have to have a icon pack installed and you have to use it in the launcher. took me a while to figure that one out too

**Comment by perceival** [comment]
Indeed. And reverting to system default icon pack unhides all the hidden apps as well.

**Comment by wrick17** [comment]
@gcladores and @perceival that was the trick. Finally got it to work. Thank you.

**Comment by lexabug** [comment]
> However... now how can I unhide the app?

Yeah, I would like to know too!

**Comment by amirzaidi** [comment]
Search in the app drawer

**Comment by gcladores** [comment]
Resolved my icon problem. I made a custom icon pack for the ones i'm missing using Iconstrutor and Apk Editor Pro. Apk Editor doesn't work on my Nexus 6P. Because it's not rooted. i had to transfer the apk form Iconstrutor to my Nexus 5 then back to 6P

**Comment by gcladores** [comment]
Would still love to have a per app icon change though. So i can change it straight from my Device storage

**Comment by mzuverink** [comment]
Why can you not offer the option of not forcing round icons? Nice launcher, makes my icons look like hell though. Im not using your launcher to mimic the look of some other device,Im here for the functionality. The option to not force round icons... Forcing round icons doesnt even 100% work. Please offer the option in settings.

**Comment by GeekyBoyHT** [comment]
You all are forgetting about - 'per app icon change from internal storage.' Please so something about that.

**Comment by pgaskin** [comment]
@amirzaidi Is there a reason why you only allow hiding apps when a custom icon pack is used? The only things I see which block hiding apps with the system icon pack are:

https://github.com/amirzaidi/Launcher3/blob/1e29b1073c1bcdb12931d3c253c7290596b62f56/src/com/google/android/apps/nexuslauncher/CustomAppFilter.java#L28-L31

https://github.com/amirzaidi/Launcher3/blob/1e29b1073c1bcdb12931d3c253c7290596b62f56/src/com/google/android/apps/nexuslauncher/CustomEditShortcut.java#L19

The only reason I see is because it overrides NexusAppFilter, which hides a few apps:
https://github.com/amirzaidi/Launcher3/blob/1e29b1073c1bcdb12931d3c253c7290596b62f56/src/com/google/android/apps/nexuslauncher/NexusAppFilter.java#L11-L23

So, is there a reason why the check for CustomEditShortcut can't be removed and then change CustomAppFilter to:

````java
public boolean shouldShowApp(ComponentName componentName, UserHandle user) {
    if (CustomIconUtils.usingValidPack(mContext)) {
        return !isHiddenApp(mContext, new ComponentKey(componentName, user));
    }
    return super.shouldShowApp(componentName, user) && !isHiddenApp(mContext, new ComponentKey(componentName, user));
}
````

This way, apps would be able to be hidden, but when using an icon pack, the system apps which are hidden by default will be able to be shown (like the behavior is right now).

**Comment by amirzaidi** [comment]
@geek1011 because I initially wanted to offer a toggle to enable/disable the Preferences option for people that do not want it, but I didn't want to have to translate it to ~60 languages so I decided to connect it to the icon pack implementation. I now realize it was a stupid idea, and I will definitely not do it like this in the future, but many people are used to it as it is. There is additional logic connected to it - when the icon pack is disabled, the default NexusAppFilter is used, and when it is enabled, icon packs are hidden to reduce app drawer clutter. Good ideas in my head, confusing UX in practice.

**Comment by Hanif-abuvani** [comment]
Lawn chair's implementation for hide app is perfect.
